### PR TITLE
Add tags to the Discourse data model

### PR DESCRIPTION
**Pull request by decentralion**
This commit updates the Discourse fetcher and mirror so that we now
retrieve the tags associated with a topic, and make them retrievable in
the mirror.

This change will not change Cred.

Test plan:
New unit tests confirm that tags are retrievable from the mirror
repository. The snapshot updates should reveal that tags are now fetched
from Discourse.

This is progress towards #2455. 

Paired with @wchargin and @topocount.

### CODE REVIEW

File: src/plugins/discourse/__snapshots__/fetch.test.js.snap [hunk 1]
```diff
@@ -324,279 +326,322 @@ Array [
...
     "title": "Welcome to Discourse",
   },
   Object {
     "bumpedMs": 1609910999621,
     "categoryId": 5,
     "id": 37,
+    "tags": Array [
+      "a-special-topic",
+      "filler-topic",
+    ],
```
Comment (decentralion) [inline src/plugins/discourse/__snapshots__/fetch.test.js.snap:340]: The fetcher now grabs tags. :)

File: src/plugins/discourse/mirrorRepository.js [hunk 2]
```diff
@@ -565,16 +613,40 @@ export class SqliteMirrorRepository
...
+          bumped_ms: topic.bumpedMs,
+          author_username: topic.authorUsername,
+        });
+
+      this._db
+        .prepare(
+          dedent`
+        DELETE FROM topic_tags
+        WHERE topic_id = :id
+      `
```
Comment (wchargin) [inline src/plugins/discourse/mirrorRepository.js:631]: Indentation extra wonky; please fix. Likewise below.

File: src/plugins/discourse/mirrorRepository.test.js [hunk 3]
```diff
@@ -7,6 +7,123 @@ import {SqliteMirrorRepository} from "./mirrorRepository";
...
+        title: "Sample topic 1",
+        timestampMs: 456789,
+        bumpedMs: 456999,
+        authorUsername: "credbot",
+      };
+      repository.addTopic(topic);
+      const actual = repository.topicById(topic.id);
+      expect(actual).toEqual(topic);
+    });
+    it("topicById gets the topic with tags", () => {
```
Comment (wchargin) [inline src/plugins/discourse/mirrorRepository.test.js:28]: nit: gets “*a* topic” with tags, not “*the* topic”, no?

File: src/plugins/discourse/mirrorRepository.test.js [hunk 4]
```diff
@@ -7,6 +7,123 @@ import {SqliteMirrorRepository} from "./mirrorRepository";
...
+      repository.addTopic(topic);
+      const actual = repository.topicById(topic.id);
+      const expected: Topic = {
+        id: 123,
+        categoryId: 42,
+        tags: ["a", "b", "c"],
+        title: "Sample topic 1",
+        timestampMs: 456789,
+        bumpedMs: 456999,
+        authorUsername: "credbot",
```
Comment (wchargin) [inline src/plugins/discourse/mirrorRepository.test.js:94]: Clearer as `{...topic, tags: ["a", "b", "c"]}`?

**Pull request merged**